pkgs/applications/networking/jdownloader2/default.nix

```
@@ -0,0 +1,71 @@
+{ stdenv, lib, fetchurl, writeScript, makeDesktopItem, jre, imagemagick, coreutils }:
+
+let
+  jddesktop = makeDesktopItem rec {
+    name = "JDownloader 2";
+    exec = wrapper;
+    icon = "jdownloader";
+    comment = "Free, open-source download management tool.";
+    desktopName = "JDownloader 2";
+    genericName = "JDownloader 2";
+    categories = "Network;";
+  };
+
+  wrapper = writeScript "jdownloader2.sh" ''
+    #! ${stdenv.shell}
+    PATH=${lib.makeBinPath [ jre coreutils ]}
+
+    JDDIR=''${XDG_DATA_HOME:-$HOME/.local/share}/jdownloader
+    JDJAR=''${JDDIR}/JDownloader.jar
+    dir=`dirname "$0"`
+
+    if [ ! -f ''${JDJAR} ]; then
+        install -Dm755 $dir/JDownloader.jar ''${JDJAR}
+    fi
+
+    ${jre}/bin/java -jar ''${JDDIR}/JDownloader.jar "''${@}"
+  '';
+
+in stdenv.mkDerivation rec {
+  name = "jdownloader2";
```

`pname` + `version` is suggested.

```suggestion
  pname = "jdownloader2";
```